library/src/main/java/com/novoda/downloadmanager/DownloadManager.java

```diff
@@ -146,20 +146,20 @@ private void executeGetAllDownloadBatchStatuses(final AllBatchStatusesCallback c
     }
 
     @Override
-    public void getDownloadStatusWithMatching(final String networkUri, final DownloadFileStatusCallback callback) {
+    public void getDownloadStatusWithMatching(final DownloadFileId downloadFileId, final DownloadFileStatusCallback callback) {
         if (downloadService == null) {
             executor.submit(
                     WaitForLockRunnable.waitFor(waitForDownloadService)
-                            .thenPerform(() -> executeFirstLocalPathForDownloadWithMatching(networkUri, callback))
+                            .thenPerform(() -> executeFirstLocalPathForDownloadWithMatching(downloadFileId, callback))
             );
         } else {
-            executeFirstLocalPathForDownloadWithMatching(networkUri, callback);
+            executeFirstLocalPathForDownloadWithMatching(downloadFileId, callback);
         }
     }
 
-    private void executeFirstLocalPathForDownloadWithMatching(String networkUri, final DownloadFileStatusCallback callback) {
+    private void executeFirstLocalPathForDownloadWithMatching(DownloadFileId downloadFileId, final DownloadFileStatusCallback callback) {
```

Comment:
no need for final, review others in this class

Reply:
300f36d